network/p2p/streams.go

```diff
@@ -92,20 +92,18 @@ func (n *streamManager) streamHandler(stream network.Stream) {
 	}
 	// no old stream
 	n.streams[stream.Conn().RemotePeer()] = stream
-	// streamHandler is supposed to be called for accepted streams, so we expect incoming here
 	incoming := stream.Conn().Stat().Direction == network.DirInbound
-	if !incoming {
-		if stream.Stat().Direction == network.DirUnknown {
-			n.log.Warnf("streamHandler: unknown direction for a steam %s to/from %s", stream.ID(), remotePeer)
-		} else {
-			n.log.Warnf("Unexpected outgoing stream in streamHandler for connection %s (%s): %s vs %s stream", stream.Conn().ID(), remotePeer, stream.Conn().Stat().Direction, stream.Stat().Direction.String())
-		}
-	}
 	n.handler(n.ctx, remotePeer, stream, incoming)
 }
 
 // Connected is called when a connection is opened
+// for both incoming (listener -> addConn) and outgoing (dialer -> addConn) connections.
 func (n *streamManager) Connected(net network.Network, conn network.Conn) {
+	if conn.Stat().Direction == network.DirInbound && !n.allowIncomingGossip {
```

Comment:
I guess outgoing is acceptable?

Reply:
there is a test checking if a node has gossip service disabled it can still send and receive gossip traffic. Another test checks if a relay node has it disabled, communication for this node will be broken.